lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1154,7 +1154,7 @@ private function getCommitOrder(array $entityChangeSet = null)
 
                 $joinColumns = reset($assoc['joinColumns']);
 
-                $calc->addDependency($targetClass->name, $class->name, (int)empty($joinColumns['nullable']));
+                $calc->addDependency($targetClass->name, $class->name, (int) ! ( ! isset($joinColumns['nullable']) || $joinColumns['nullable'] === true));
```

Comment:
Let's keep it on integration tests for now - unit testing the UoW is a bit more messy, and we can add that just before merging, rather than early during the fixing process